### PR TITLE
Refine phone number -- AB#3131

### PR DESCRIPTION
**Pull request by olivierroyESDC**
### Description
Refined phone number page, added an alternate phone number field. Updated text with the figma..
Upon submit, sends the user right back to the personal-information page and not the confirm page.

### Related Azure Boards Work Items
[AB#3131](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/3131)

### Checklist

- [x] I have tested the changes locally
- [x] I have updated the documentation if necessary
- [x] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [ ] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/__tests__/routes/_protected+/personal-information+/phone-number+/edit.test.tsx [hunk 1]
```diff
@@ -43,6 +41,13 @@ vi.mock('~/utils/locale-utils.server', () => ({
 }));
 vi.mock('~/utils/env.server', () => ({
   featureEnabled: vi.fn().mockResolvedValue(true),
+  getEnv: vi.fn().mockReturnValue({
+    INTEROP_CCT_API_BASE_URI: 'https://api.example.com',
+    INTEROP_CCT_API_SUBSCRIPTION_KEY: '00000000000000000000000000000000',
+    INTEROP_CCT_API_COMMUNITY: 'CDCP',
+    ENGLISH_LANGUAGE_CODE: 1033,
+    FRENCH_LANGUAGE_CODE: 1036,
```
Comment (nicholas-ly) [inline frontend/__tests__/routes/_protected+/personal-information+/phone-number+/edit.test.tsx:49]: No need to mock these particular environment variables. They aren't being used for the phone numbers feature, only letters.

File: frontend/__tests__/routes/_protected+/personal-information+/phone-number+/edit.test.tsx [hunk 2]
```diff
@@ -67,7 +70,6 @@ describe('_gcweb-app.personal-information.phone-number.edit', () => {
 
       expect(data).toMatchObject({
         meta: {},
-        userInfo: { id: 'some-id', phoneNumber: '(111) 222-3333' },
       });
```
Comment (nicholas-ly) [inline frontend/__tests__/routes/_protected+/personal-information+/phone-number+/edit.test.tsx:73]: This unit test passes but does not perform the correct assertion. The user's phone number should be found and returned as part of the `loader()`.

File: frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/edit.tsx [hunk 3]
```diff
@@ -39,36 +45,34 @@ export const meta: MetaFunction<typeof loader> = mergeMeta(({ data }) => {
...
-    throw new Response(null, { status: 404 });
-  }
-
   const csrfToken = String(session.get('csrfToken'));
+  const personalInformationRouteHelper = getPersonalInformationRouteHelpers();
 
+  const userInfoToken: UserinfoToken = session.get('userInfoToken');
+  const personalInformation = await personalInformationRouteHelper.getPersonalInformation(userInfoToken, params, request, session);
+
+  session.set('personalInformation', personalInformation);
```
Comment (nicholas-ly) [inline frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/edit.tsx:62]: I wonder if we should be setting the `personalInformation` in the session here. At this point, `personalInformation` should already be in the session from a previous request in my opinion. 

Maybe we should be setting the `personalInformation` in the session in the `/personal-information` index page instead. Then, if `personalInformation` cannot be found on this edit page, we redirect to `/personal-information`.

File: frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/edit.tsx [hunk 4]
```diff
@@ -39,36 +45,34 @@ export const meta: MetaFunction<typeof loader> = mergeMeta(({ data }) => {
...
+  return json({ csrfToken, meta, personalInformation });
 }
 
 export async function action({ context: { session }, params, request }: ActionFunctionArgs) {
   const log = getLogger('phone-number/edit');
-
+  const personalInformationService = await getPersonalInformationService();
   const instrumentationService = getInstrumentationService();
   const raoidcService = await getRaoidcService();
+  const personalInformation: PersonalInfo = session.get('personalInformation');
```
Comment (nicholas-ly) [inline frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/edit.tsx:75]: We should be calling the `personalInformationRouteHelper.getPersonalInformation(..)` function to get the `personalInformation` instead.

File: frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/edit.tsx [hunk 5]
```diff
@@ -77,6 +81,20 @@ export async function action({ context: { session }, params, request }: ActionFu
...
+          } else {
+            return isValidPhoneNumber(val, 'CA');
+          }
+        },
+        { message: 'invalid-phone-format' },
+      )
+      .optional(),
   });
 
   const formData = Object.fromEntries(await request.formData());
```
Comment (nicholas-ly) [inline frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/edit.tsx:100]: Can we change the use of `Object.fromEntries(..)` in favour of using `formData = await request.formData();` to be consistent with the rest of the application's `action(..)`?

File: frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/edit.tsx [hunk 6]
```diff
@@ -97,24 +115,31 @@ export async function action({ context: { session }, params, request }: ActionFu
...
   const errorSummaryId = 'error-summary';
 
   const { t } = useTranslation(handle.i18nNamespaces);
 
   const defaultValues = {
-    phoneNumber: actionData?.formData.phoneNumber ?? userInfo.phoneNumber ?? '',
+    phoneNumber: personalInformation.primaryTelephoneNumber ?? '',
+    alternatePhoneNumber: personalInformation.alternateTelephoneNumber ?? '',
   };
+  t;
```
Comment (nicholas-ly) [inline frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/edit.tsx:142]:
```suggestion
```

File: frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/edit.tsx [hunk 7]
```diff
@@ -147,21 +173,27 @@ export default function PhoneNumberEdit() {
...
     <>
-      <p className="mb-8 border-b border-gray-200 pb-8 text-lg text-gray-500">{t('personal-information:phone-number.edit.subtitle')}</p>
       {errorSummaryItems.length > 0 && <ErrorSummary id={errorSummaryId} errors={errorSummaryItems} />}
       <Form method="post" noValidate>
         <input type="hidden" name="_csrf" value={csrfToken} />
         <div className="my-6">
-          <p className="mb-4 text-red-600">{t('gcweb:asterisk-indicates-required-field')}</p>
-          <InputField id="phoneNumber" name="phoneNumber" type="tel" label={t('personal-information:phone-number.edit.component.phone')} required defaultValue={defaultValues.phoneNumber} errorMessage={errorMessages.phoneNumber} />
+          <InputField id="phoneNumber" name="phoneNumber" type="tel" label={t('personal-information:phone-number.edit.component.phone')} defaultValue={defaultValues.phoneNumber} errorMessage={errorMessages.phoneNumber} />
+          <InputField
```
Comment (nicholas-ly) [inline frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/edit.tsx:181]: The "Alternate telephone number" label is too close to the input field above it:
image

Consider adding some spacing between these two elements.

File: frontend/__tests__/routes/_protected+/personal-information+/phone-number+/edit.test.tsx (hunk 1 above)
Comment (olivierroyESDC) [reply frontend/__tests__/routes/_protected+/personal-information+/phone-number+/edit.test.tsx]: removed

File: frontend/__tests__/routes/_protected+/personal-information+/phone-number+/edit.test.tsx (hunk 2 above)
Comment (olivierroyESDC) [reply frontend/__tests__/routes/_protected+/personal-information+/phone-number+/edit.test.tsx:69]: i changed the expected data.

File: frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/edit.tsx (hunk 6 above)
Comment (olivierroyESDC) [reply frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/edit.tsx]: removed

File: frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/edit.tsx (hunk 7 above)
Comment (olivierroyESDC) [reply frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/edit.tsx:183]: space added

File: frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/edit.tsx (hunk 3 above)
Comment (nicholas-ly) [reply frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/edit.tsx]: As discussed, the helper function already redirects the user to `/data-unavailable` if `personalInformation` does not exist in the session or service. 

We can move the `session.set('personalInformation', personalInformation) line in the route helper if `personalInformation` is found from the service.

File: frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/edit.tsx (hunk 4 above)
Comment (olivierroyESDC) [reply frontend/app/routes/$lang+/_protected+/personal-information+/phone-number+/edit.tsx]: done

**Review by nicholas-ly — APPROVED** [review]

**Review by nicholas-ly — DISMISSED** [review]

**Comment by gregory-j-baker** [comment]
```
npm run format:check && npm run lint && npm run typecheck && npm run test:unit -- run && npm run test:e2e
```

Run this to find all the possible build issues. 😄

**Review by nicholas-ly — APPROVED** [review]

**Pull request merged**